runtime/codert_vm/cnathelp.cpp

```diff
@@ -1490,17 +1490,19 @@ old_fast_jitCheckCastForArrayStore(J9VMThread *currentThread)
 {
 	void *slowPath = NULL;
 	OLD_JIT_HELPER_PROLOGUE(2);
-	DECLARE_JIT_CLASS_PARM(castClass, 1);
+	DECLARE_JIT_CLASS_PARM(castClassArray, 1);	// <-- castClassArray should be the array class, not its base class
 	DECLARE_JIT_PARM(j9object_t, object, 2);
-	/* null can be cast to anything, except if castClass is a primitive VT */
+	Assert_CodertVM_true(J9CLASS_IS_ARRAY(castClassArray));
+	/* null can be cast to anything, except if castClassArray is a null-restricted array */
 	if (NULL != object) {
 		J9Class *instanceClass = J9OBJECT_CLAZZ(currentThread, object);
+		J9Class *castClass = ((J9ArrayClass*)castClassArray)->leafComponentType;
```

Comment:
Should this be looking at the `componentType` rather than the `leafComponentType`?